parsec/core/fs/storage/workspace_storage.py

```diff
@@ -64,15 +57,126 @@ def __init__(
         self.fd_counter = 0
 
         # Locking structures
-        self.locking_tasks = {}
-        self.entry_locks = defaultdict(trio.Lock)
+        self.locking_tasks: Dict[EntryID, hazmat.Task] = {}
+        self.entry_locks: Dict[EntryID, trio.Lock] = defaultdict(trio.Lock)
 
         # Manifest and block storage
+        self.block_storage = block_storage
+        self.chunk_storage = chunk_storage
+
+    def _get_next_fd(self) -> FileDescriptor:
+        self.fd_counter += 1
+        return FileDescriptor(self.fd_counter)
+
+    async def get_manifest(self, entry_id: EntryID) -> LocalManifest:
+        raise NotImplementedError
+
+    # Locking helpers
+
+    @asynccontextmanager
+    async def lock_entry_id(self, entry_id: EntryID):
+        async with self.entry_locks[entry_id]:
+            try:
+                self.locking_tasks[entry_id] = hazmat.current_task()
+                yield entry_id
+            finally:
+                del self.locking_tasks[entry_id]
+
+    @asynccontextmanager
+    async def lock_manifest(self, entry_id: EntryID):
+        async with self.lock_entry_id(entry_id):
+            yield await self.get_manifest(entry_id)
+
+    def _check_lock_status(self, entry_id: EntryID) -> None:
+        task = self.locking_tasks.get(entry_id)
+        if task != hazmat.current_task():
+            raise RuntimeError(f"Entry `{entry_id}` modified without beeing locked")
+
+    # File management interface
+
+    def create_file_descriptor(self, manifest: LocalFileManifest) -> FileDescriptor:
+        assert isinstance(manifest, LocalFileManifest)
```

Comment:
asset still needed ?

Reply:
Maybe non but mypy does not cover the entire code base, so it might be useful to keep it.